handler/proxy.go

```diff
@@ -209,6 +221,12 @@ func (p *Proxy) roundtrip(rw http.ResponseWriter, req *http.Request) {
 		outreq.Header.Set("X-Forwarded-For", clientIP)
 	}
 
+	if ReClientSupportsGZ.MatchString(req.Header.Get(AcceptEncodingHeader)) {
+		req.Header.Set(AcceptEncodingHeader, GzipName)
+	} else {
+		req.Header.Del(AcceptEncodingHeader)
+	}
+
 	res, err := p.transport.RoundTrip(outreq)
```

Comment:
As already communicated, we should check the proxy transport option `DisableCompression`.